docs/io/ascii/read.rst

```diff
@@ -577,6 +577,15 @@ The default converters for each column can be overridden with the
   ...               'col2': [ascii.convert_numpy(np.float32)]}
   >>> ascii.read('file.dat', converters=converters)  # doctest: +SKIP
 
+In addition to single column names you can use wildcards via `fnmatch` to
+select multiple columns. For example, we can set the format for all columns
+where the column name starts with i to `np.unit` while applying default
+converters to all other columns in the table::
+
+  >>> import numpy as np
+  >>> converters = {'i*': [ascii.convert_numpy(np.uint)]}
```

Comment:
TBH, this isn't intuitive to me. I would expect the key to be column names, as in `{'*': [ascii.convert_numpy(np.uint)]}` would make all columns integers.

I am not sure of the value of adding this feature. 👀

Reply:
`{'*': [ascii.convert_numpy(np.uint)]}` would make all columns integers, but `{'i*': [ascii.convert_numpy(np.uint)]}` would only make columns integers that start with i, as in unis `ls i*`.

Reply:
But I'd be happy to go with #11889 instead of this one...

Reply:
Sure, but as a user, naturally, I would want to map by column names, because I might know, say, "ID" should be int and "Flux" should be float, but I won't know what `Table` will map them to beforehand.

Reply:
So using wildcard would be like `{'flux*': [ascii.convert_numpy(float)]}` or something.

But then again, this is just theory now. I don't even quite recall what the actual use case was that caused me to request the feature all those years ago.

Reply:
What about making the doc example match the use case in #4934 in which you have an ASCII table of integer-like values but you want all the columns to be float?  So `{'*': [ascii.convert_numpy(float)]}`.

Reply:
`{'*': [ascii.convert_numpy(float)]}` is easier to remember that using `i*`. But then again, in all these years, no one else asked for this feature and I don't need it anymore. So... is this worth adding now? 🤷

Reply:
OK, I see that I have chosen a bad example. `{'*abc*': [ascii.convert_numpy(float)]}` would convert the following columns to float: aabcc, 1abc234, abc, but not aabb or 123bc34.

Reply:
Ah, so that is the column name? I can't tell because I just assumed it is a continuation of the previous example:

```python
>>> import numpy as np
>>> converters = {'col1': [ascii.convert_numpy(np.uint)],
...               'col2': [ascii.convert_numpy(np.float32)]}
>>> ascii.read('file.dat', converters=converters)  # doctest: +SKIP
``` 

Maybe the doc can be changed to use `{'col*': [ascii.convert_numpy(float)]}` instead?

Reply:
Yes, I think just using `col*` in the example would solve the confusion!